### PR TITLE
Syncing cqgi.py back up with dcowden/master.

### PR DESCRIPTION
**Pull request by jmwright**
This is in response to #85 . This updates cqgi.py to the version that was updated in dcowden/master after the fork.

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
@jmwright thanks! Is this WIP? I see some CQGI test cases failing.

**Comment by jmwright** [comment]
Yeah, sorry. WIP. I totally missed changing build_object to show_object in the test script and then lost it in the output from the other tests.

**Comment by jmwright** [comment]
@adam-urbanczyk This should be ready to merge. All the CQGI tests pass.

The documentation has been updated to change `build_object` to `show_object` in the separate documentation PR for master. Not sure if that was the best branch to do that PR on.

File: cadquery/cqgi.py [hunk 1]
```diff
@@ -103,25 +101,23 @@ def build(self, build_parameters=None, build_options=None):
...
-            else:
-                raise NoOutputError(
-                    "Script did not call build_object-- no output available.")
+            exec (c, env)
+            result.set_debug(collector.debugObjects )
+            result.set_success_result(collector.outputObjects)
+
         except Exception as ex:
-            print("Error Executing Script:")
+            #print "Error Executing Script:"
```
Comment (adam-urbanczyk) [inline cadquery/cqgi.py]: Are the commented lines really needed?

File: cadquery/cqgi.py [hunk 2]
```diff
@@ -103,25 +101,23 @@ def build(self, build_parameters=None, build_options=None):
...
             result.set_failure_result(ex)
-            traceback.print_exc()
-            print("Full Text of Script:")
-            print(self.script_source)
+            #traceback.print_exc()
```
Comment (adam-urbanczyk) [inline cadquery/cqgi.py]: Same here

File: cadquery/cqgi.py [hunk 3]
```diff
@@ -177,15 +180,14 @@ class ScriptMetadata(object):
...
     def __init__(self):
         self.parameters = {}
 
     def add_script_parameter(self, p):
         self.parameters[p.name] = p
 
-    def add_parameter_description(self, name, description):
-        print('Adding Parameter name=%s, desc=%s' % (name, description))
+    def add_parameter_description(self,name,description):
+        #print 'Adding Parameter name=%s, desc=%s' % ( name, description )
```
Comment (adam-urbanczyk) [inline cadquery/cqgi.py]: another occurance

File: cadquery/cqgi.py [hunk 4]
```diff
@@ -414,33 +413,30 @@ def add_entry(self, name, value):
...
                 # first parameter is the variable,
                 # second is the description
                 varname = node.args[0].id
                 desc = node.args[1].s
-                self.cqModel.add_parameter_description(varname, desc)
+                self.cqModel.add_parameter_description(varname,desc)
 
-        except:
-            print("Unable to handle function call")
+       except:
```
Comment (adam-urbanczyk) [inline cadquery/cqgi.py]: Do we want too catch all exceptions here?

File: cadquery/cqgi.py [hunk 5]
```diff
@@ -466,12 +463,12 @@ def handle_assignment(self, var_name, value_node):
...
-                 else:
-                     self.cqModel.add_script_parameter(
-                         InputParameter.create(value_node, var_name, BooleanParameterType, False))
+                if value_node.value == True:
+                    self.cqModel.add_script_parameter(
+                        InputParameter.create(value_node, var_name, BooleanParameterType, True))
+                else:
+                    self.cqModel.add_script_parameter(
+                        InputParameter.create(value_node, var_name, BooleanParameterType, False))
         except:
```
Comment (adam-urbanczyk) [inline cadquery/cqgi.py]: Same question as above

File: cadquery/cqgi.py [hunk 6]
```diff
@@ -497,7 +495,6 @@ def visit_Assign(self, node):
                     self.handle_assignment(n.id, v)
         except:
```
Comment (adam-urbanczyk) [inline cadquery/cqgi.py]: One more. NB: I am not saying this is wrong - just wondering

**Comment by adam-urbanczyk** [comment]
@jmwright  thanks for this work. I added some questions. Up to you if actions are required.

CI as expected fails on Mac. I think I want to merge nevertheless and handle Mac as a separate issue (which BTW might take long to get solved given the limited access to Mac HW). Do you agree?

**Comment by jmwright** [comment]
@adam-urbanczyk Thanks for the feedback. I can remove the comments no problem, but I'd have to make an educated guess on the original intent of the `except` blocks. @dcowden do you remember what your thought process was on the code from the last 3 questions involving the `except` clauses? 

> I think I want to merge nevertheless and handle Mac as a separate issue (which BTW might take long to get solved given the limited access to Mac HW). Do you agree?

I think that would be fine once the questions above are resolved. I'm hoping to get some visibility into the OSX segfault, but it might take some time to sort it all out.

**Comment by adam-urbanczyk** [comment]
@jmwright thanks, merging your PR.

**Pull request merged**